go/types/list_editor.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/attic-labs/noms/go/d"
+)
+
+type ListEditor struct {
+	l     List
+	edits *listEdit
+}
+
+func NewListEditor(l List) *ListEditor {
+	return &ListEditor{l, nil}
+}
+
+func (le *ListEditor) Kind() NomsKind {
+	return ListKind
+}
+
+func (le *ListEditor) Value(vrw ValueReadWriter) Value {
+	return le.List(vrw)
+}
+
+func (le *ListEditor) List(vrw ValueReadWriter) List {
+	if le.edits == nil {
+		return le.l // no edits
+	}
+
+	seq := le.l.sequence()
+	vr := seq.valueReader()
+
+	if vrw != nil {
+		vr = vrw
+	}
+
+	cursChan := make(chan chan *sequenceCursor)
+	spliceChan := make(chan chan listEdit)
+
+	go func() {
+		for edit := le.edits; edit != nil; edit = edit.next {
+			edit := edit
+
+			// TODO: Use ReadMany
+			cc := make(chan *sequenceCursor, 1)
+			cursChan <- cc
+
+			go func() {
+				cc <- newCursorAtIndex(seq, edit.idx, false)
+			}()
+
+			sc := make(chan listEdit, 1)
+			spliceChan <- sc
+
+			wg := sync.WaitGroup{}
+			subEditors := false
+			for i, v := range edit.inserted {
+				if _, ok := v.(Value); ok {
```

Comment:
No. Basically, if it's a Valuable, then it's an Editor and we need to build it in its own thread, if it's a Value, then we can just pass it along (because it's already built).